orion/pkg/repos/user_repo.go

```diff
@@ -29,15 +29,27 @@ func (ur *userRepo) Initialize(db *sql.DB) {
 	ur.db = db
 }
 
-func (ur *userRepo) SelectAll() ([]domains.User, error) {
+func (ur *userRepo) SelectAll(search string, pageSize, offset int) ([]domains.User, error) {
 	results := make([]domains.User, 0)
 
-	stmt, err := ur.db.Prepare("SELECT * FROM users")
+	var query string
+	if len(search) == 0 {
+		query = "SELECT * FROM users LIMIT ? OFFSET ?"
+	} else {
+		query = "SELECT * FROM users WHERE ? IN (first_name, last_name, middle_name) LIMIT ? OFFSET ?"
+	}
+	stmt, err := ur.db.Prepare(query)
 	if err != nil {
 		return nil, err
 	}
 	defer stmt.Close()
-	rows, err := stmt.Query()
+
+	var rows *sql.Rows
+	if len(search) == 0 {
```

Comment:
nit: you duplicate logic here and line 36 (`len(search == 0)`)
You could define a boolean value above and use it both here and line the if/else block in 36.